proteus/mprans/SpatialTools.py

```diff
@@ -1390,7 +1385,7 @@ def assembleAuxiliaryVariables(domain):
     -----
     Should be called after assembleGeometry
     """
-    domain.auxiliaryVariables = []
+    domain.auxiliaryVariables = {'twp': [], 'vof': []}
```

Comment:
Yes, thanks for spotting this, there is an error that was introduced in the latest commit: it should be `aux['twp'][-1]`. The `-1` here is to access the latest auxiliary variable added in the list (a class instance of `RigidBody` in this case). But maybe it looks a bit clumsy anyway, we could do something clearer like this for the same effect:
```
body = RigidBody(shape)
aux['twp'] += [body]
body.i_start = start_flag+1
...
```

Because the auxiliary variables are already stored in a dictionary after calling `assembleAuxiliaryVariables`, each `*_n.py` files would just need something like: 
```
auxiliaryVariables = domain.auxiliaryVariables['twp']  # this should return a list
```
New keys such as `redist` or `ls` can be added to the `domain.auxiliaryVariables` dictionary that is initialised in the function `assembleAuxiliaryVariables(domain)`. I added vof there because I was using it for column gauges.